release/train_tests/benchmark/train_benchmark.py

```diff
@@ -57,22 +70,28 @@ def run(self):
                 self.validate_and_checkpoint()
 
             if ray.train.get_context().get_world_rank() == 0:
-                pprint.pprint(self.get_metrics())
+                logger.info(pprint.pformat(self.get_metrics(), indent=2))
 
     def train_epoch(self):
         with self._metrics["train/epoch"].timer():
             self._train_epoch()
 
     def _train_epoch(self):
         if ray.train.get_context().get_world_rank() == 0:
-            print(f"Training starting @ epoch={self._train_epoch_idx}")
+            logger.info(f"[Train] Starting @ epoch={self._train_epoch_idx}")
+
+        train_dataloader = self.factory.get_train_dataloader()
 
         # NOTE: Time the first batch separately since it includes the dataset
         # pipeline warmup time.
         with self._metrics["train/iter_first_batch"].timer():
-            batch = self.get_next_batch(self.train_dataloader)
+            batch = self.get_next_batch(train_dataloader)
 
-        # TODO: Handle the case where we restored to a middle of the epoch.
+        # Skip through batches if we restored to a middle of the epoch.
+        # TODO: Compare this baseline to the data checkpointing approach once we have it.
+        for _ in range(self._train_batch_idx):
+            with self._metrics["train/iter_skip_batch"].timer():
+                self.get_next_batch(train_dataloader)
```

Comment:
Keeping an accurate total for the time spent "skipping" batches is a little tricky because this step could take a long time (ex: if training was killed previously at the second to last batch). Then, a node could get killed in this time, and the time spent skipping batches would be lost from the metrics, which are only snapshotted every checkpoint.

May want to checkpoint the metrics separately from the model if we want to track this better.